gax-java/gax/src/main/java/com/google/api/gax/tracing/MetricsTracerFactory.java

```diff
@@ -45,13 +47,31 @@
 public class MetricsTracerFactory implements ApiTracerFactory {
   protected MetricsRecorder metricsRecorder;
 
+  // These are client attributes and pertain every single trace
+  private final Map<String, String> attributes;
+
+  /** Creates a MetricsTracerFactory with no additional client level attributes. */
   public MetricsTracerFactory(MetricsRecorder metricsRecorder) {
+    this(metricsRecorder, ImmutableMap.of());
+  }
+
+  /**
+   * Pass in a Map of client level attributes which will be added to every single MetricsTracer
+   * created from the ApiTracerFactory.
+   */
+  public MetricsTracerFactory(MetricsRecorder metricsRecorder, Map<String, String> attributes) {
     this.metricsRecorder = metricsRecorder;
+    this.attributes = attributes;
   }
```

Comment:
Additional constructor to take in user configured in client level attributes. 

In the future, we can have Gax automatically collect attributes and this would need a constructor to take in a set of attribute flags. Class is marked with InternalApi and could be open to more refactoring if we want to pursue this.